src/main/java/com/minecolonies/coremod/util/InventoryFunctions.java

```diff
@@ -53,89 +50,110 @@ public static boolean matchFirstInInventory(
-     *
-     * @param inventory the inventory to search in
-     * @param tester    the function to use for testing slots
-     * @param action    the function to use if a slot matches
-     * @return true if it found a stack
-     */
-    public static boolean matchFirstInInventoryWithInventory(
-            final IInventory inventory, @NotNull final Predicate<ItemStack> tester,
-            @NotNull final Consumer<Tuple<Integer, IInventory>> action)
-    {
-        return matchFirstInInventory(inventory, inv -> slot -> stack ->
-        {
-            if (tester.test(stack))
-            {
-                action.accept(new Tuple<>(slot, inventory));
-                return true;
-            }
-            return false;
-        });
-    }
-
-    /**
      * Topmost matchFirst function, will stop after it finds the first itemstack.
      *
-     * @param inventory the inventory to search in
-     * @param tester    the function to use for testing slots
+     * @param provider the provider to search in
+     * @param tester   the function to use for testing slots
      * @return true if it found a stack
      */
-    private static boolean matchFirstInInventory(
-                                                  final IInventory inventory, @NotNull final Function<IInventory, Function<Integer,
-                                                                                                                            Predicate<ItemStack>>> tester)
+    private static boolean matchFirstInProvider(
+                                                 final ICapabilityProvider provider, @NotNull final Function<ICapabilityProvider, Function<Integer,
+                                                                                                                                            Predicate<ItemStack>>> tester)
     {
-        return matchInInventory(inventory, tester, true);
+        return matchInProvider(provider, tester, true);
     }
 
     /**
-     * Topmost function to actually loop over the inventory.
+     * Topmost function to actually loop over the provider.
      * Will return if it found something.
      *
-     * @param inventory      the inventory to loop over
+     * @param provider       the provider to loop over
      * @param tester         the function to use for testing slots
      * @param stopAfterFirst if it should stop executing after finding one stack that applies
      * @return true if it found a stack
      */
-    private static boolean matchInInventory(
-                                             @Nullable final IInventory inventory,
-                                             @NotNull final Function<IInventory, Function<Integer,
-                                                                                           Predicate<ItemStack>>> tester,
-                                             final boolean stopAfterFirst)
+    private static boolean matchInProvider(
```

Comment:
I donnot see the the use of these anyway and would like to refactor them if possible. I think we use it ones to move itemstacks over that match the predicate btw.